src/Pim/Bundle/CatalogBundle/tests/integration/ProductValidation/UniqueIdentifierValidationIntegration.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace tests\integration\Pim\Bundle\CatalogBundle\ProductValidation;
+
+use Akeneo\Component\StorageUtils\Saver\SaverInterface;
+use Akeneo\Test\Integration\Configuration;
+use Akeneo\Test\Integration\TestCase;
+use Pim\Component\Catalog\Builder\ProductBuilderInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * @author    Damien Carcel (damien.carcel@akeneo.com)
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class UniqueIdentifierValidationIntegration extends TestCase
+{
+    /** @var SaverInterface */
+    protected $productSaver;
+
+    /** @var ValidatorInterface */
+    protected $productValidator;
+
+    /** @var ProductBuilderInterface */
+    protected $productBuilder;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->productBuilder = $this->get('pim_catalog.builder.product');
```

Comment:
Be careful when making your test case stateful, as we don't know exactly how phpunit works we could have surprises (such as memory leaks). Maybe just use protected methods? (`createProduct`, `validateProduct`...)